frontend/simple/js/state.js

```diff
@@ -92,6 +92,26 @@ const getters = {
 }
 
 const actions = {
+  async login (
+    {dispatch, commit, state}: {dispatch: Function, commit: Function, state: Object},
+    user: string
+  ) {
+    commit('login', user)
+    await dispatch('loadSettings')
+    await db.saveCurrentUser(user)
+    for (let key of Object.keys(state.contracts)) {
+      await backend.subscribe(key)
+    }
+    Vue.events.$emit('login', user)
+  },
+  async logout (
+    {dispatch, commit, state}: {dispatch: Function, commit: Function, state: Object},
+  ) {
+    await dispatch('saveSettings')
+    await db.clearCurrentUser()
+    commit('logout')
+    Vue.events.$emit('logout')
```

Comment:
Logout should be the inverse of login. So since we subscribed in login, we should unsubscribe in logout.